ui/hooks/snaps/useTransactionInsightSnaps.js

```diff
@@ -11,38 +11,26 @@ export function useTransactionInsightSnaps({
   chainId,
   origin,
   insightSnaps,
-  eagerFetching = true,
-  insightSnapId = '',
 }) {
   const subjects = useSelector(getPermissionSubjectsDeepEqual);
 
   const [loading, setLoading] = useState(true);
   const [data, setData] = useState(undefined);
-  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
-  const [hasFetchedV2Insight, setHasFetchedV2Insight] = useState(false);
-  ///: END:ONLY_INCLUDE_IF
+  const [hasFetchedInsight, setHasFetchedInsight] = useState(false);
+
   useEffect(() => {
     let cancelled = false;
 
     async function fetchInsight() {
```

Comment:
Wondering if we can make this hook generic and dedupe the code between `useSignatureInsights` and `useTransactionInsightSnaps`?

Reply:
Hmm, perhaps? I'm worried about future insight features and this hook becoming bloated.